docs/architecture/incremental-dedupted-story.md

```diff
@@ -0,0 +1,131 @@
+
+## Rules
+
+As mentioned above, the delta from a sync will be _appended_ to the existing story data in the data warehouse. In addition, it will update the associated record in the final table.
+Let's walk through a few examples.
+
+### Newly Created Record
+
+Assume that `updated_at` is our `cursor_field` and `name` is the `primary_key`. Let's say the following data already exists into our data warehouse.
+
+```javascript
+[
+    { "name": "Louis XVI", "deceased": false, "updated_at":  1754 },
+    { "name": "Marie Antoinette", "deceased": false, "updated_at":  1755 }
+]
+```
+
+In the next sync, the delta contains the following record:
+
+```javascript
+    { "name": "Louis XVII", "deceased": false, "updated_at": 1785 }
+```
+
+At the end of this incremental sync, the data warehouse would now contain:
+
+```javascript
+[
+    { "name": "Louis XVI", "deceased": false, "updated_at":  1754 },
+    { "name": "Marie Antoinette", "deceased": false, "updated_at":  1755 },
+    { "name": "Louis XVII", "deceased": false, "updated_at": 1785 }
+]
+```
+
+### Updating a Record
+
+Let's assume that our warehouse contains all the data that it did at the end of the previous section. Now unfortunately the king and queen lose their heads. Let's see that delta:
+
+```javascript
+[
+    { "name": "Louis XVI", "deceased": true, "updated_at": 1793 },
+    { "name": "Marie Antoinette", "deceased": true, "updated_at": 1793 }
+]
+```
+
+The output we expect to see in the warehouse is as follows:
+
+In the story table:
+```javascript
+[
+    { "name": "Louis XVI", "deceased": false, "updated_at":  1754, "start_at": 1754, "end_at": 1793 },
+    { "name": "Louis XVI", "deceased": true, "updated_at": 1793, "start_at": 1793, "end_at": NULL },
+
+    { "name": "Louis XVII", "deceased": false, "updated_at": 1785, "start_at": 1785, "end_at": NULL }
+
+    { "name": "Marie Antoinette", "deceased": false, "updated_at":  1755, "start_at": 1755, "end_at": 1793 },
+    { "name": "Marie Antoinette", "deceased": true, "updated_at": 1793, "start_at: 1793, "end_at": NULL }
+]
+```
+
+In the final de-duplicated table:
+```javascript
+[
+    { "name": "Louis XVI", "deceased": true, "updated_at": 1793 },
+    { "name": "Louis XVII", "deceased": false, "updated_at": 1785 },
+    { "name": "Marie Antoinette", "deceased": true, "updated_at": 1793 }
+]
+```
+
+## Source-Defined Cursor
+
+Some sources are able to determine the cursor that they use without any user input. For example, in the [exchange rates source](../integrations/sources/exchangeratesapi-io.md), the source knows that the date field should be used to determine the last record that was synced. In these cases, simply select the incremental option in the UI.
+
+![](../.gitbook/assets/incremental_source_defined.png)
+
+\(You can find a more technical details about the configuration data model [here](catalog.md)\).
+
+## User-Defined Cursor
```

Comment:
there is a lot of duplication between this and incremental append in explaining the cursor stuff. do you think it would be worthwhile trying to deduplicate it? maybe in this one just link to the incremental append sections that describe cursor fields? 

just an idea, i'm not feeling too sure. i dislike the duplication but at the same time, having to look at multiple docs is pretty annoying too.

Reply:
Should the two modes be described on the same page then? since they are part of the same behavior "family" with slight changes?